setup.py

```diff
@@ -122,31 +176,73 @@ def run_setup():
         'Programming Language :: Python :: Implementation :: PyPy',
         'Topic :: Scientific/Engineering :: Mathematics',
         'Topic :: Software Development :: Libraries :: Python Modules' ],
-      python_requires='>=3.6',
-      install_requires=[
-          'ply',
-      ],
-      packages=find_packages(exclude=("scripts",)),
-      package_data={
-          "pyomo.contrib.appsi.cmodel": ["src/*"],
-          "pyomo.contrib.mcpp": ["*.cpp"],
-          "pyomo.contrib.pynumero": ['src/*', 'src/tests/*'],
-          "pyomo.contrib.viewer": ["*.ui"],
-      },
-      #include_package_data=True,
-      ext_modules = ext_modules,
-      entry_points="""
-        [console_scripts]
-        pyomo = pyomo.scripting.pyomo_main:main_console_script
-
-        [pyomo.command]
-        pyomo.help = pyomo.scripting.driver_help
-        pyomo.viewer=pyomo.contrib.viewer.pyomo_viewer
-      """
-      )
+    python_requires = '>=3.6',
+    install_requires = [
+        'ply',
+    ],
+    extras_require = {
+        'tests': [
+            'coverage',
+            'nose',
+            'parameterized',
+            'pybind11',
+        ],
+        'docs': [
+            'Sphinx>2',
+            'sphinx-copybutton',
+            'sphinx_rtd_theme>0.5',
+            'sphinxcontrib-jsmath',
+            'sphinxcontrib-napoleon',
+            'numpy', # Needed by autodoc for pynumero
+            'scipy', # Needed by autodoc for pynumero
+        ],
+        'optional': [
+            'dill',      # No direct use, but improves lambda pickle
+            'ipython',   # contrib.viewer
+            'networkx',  # network, incidence_analysis, community_detection
+            'openpyxl',  # dataportals
+            #'pathos',   # requested for #963, but PR currently closed
+            'pint',      # units
+            #'pyro4',    # used by PySP; no longer needed by core Pyomo
```

Comment:
Is it worth leaving this in here if it's not actually required and there aren't plans for it to be?

Reply:
Nope.  Removed.